### ISSUE TITLE
No instructions and no result?

### ISSUE THREAD

**Issue by MostHated** [milesburchell/RyzenMasterVBSFix#1]
Hey there,
    There are no instructions for this and when I run it, there are no results either, error or otherwise. Is there something specific about how to execute it? I see the path in the code to the location of RyzenMaster is correct.

Thanks,
-MH

**Comment by milesburchell** [comment]
If you run the program from an elevated command prompt, passing the path to
your .exe as an argument, you will be able to see any output in the console
window. Compiling in Debug mode will enable further debug output.

**Comment by MostHated** [comment]
Unfortunately, running in an elevated terminal provided no additional output.

```terminal
❯ cd 'C:\Program Files\AMD\RyzenMaster\bin'

AMD/RyzenMaster/bin 
❯ .\VBSFix.exe '.\AMD Ryzen Master.exe'

AMD/RyzenMaster/bin 
❯ .\VBSFix.exe 'C:\Program Files\AMD\RyzenMaster\bin\AMD Ryzen Master.exe'

AMD/RyzenMaster/bin 
❯ 
```

I have not tried building with debug. Perhaps I can give that a go later.

**Comment by macaquinyho** [comment]
Same issue, doesn't work at least with the precompiled exe. 
EDIT: I copied it to he install folder and ran it with powershell and it gives me this:
`FindPattern failed.`
I have the lastest Ryzen Master version

**Comment by MostHated** [comment]
I, as well, was able to finally get ```FindPattern failed.``` after messing around with things. I attempted to do the now listed Image File Execution attachment as a debugger and received the message.

---
Edit - I just attempted to run via Visual Studio and received the output of the following:

```
'VBSFix.exe' (Win32): Loaded 'E:\GitHub\RyzenMasterVBSFix\x64\Debug\VBSFix.exe'. Symbols loaded.
'VBSFix.exe' (Win32): Loaded 'C:\Windows\System32\ntdll.dll'. 
'VBSFix.exe' (Win32): Loaded 'C:\Windows\System32\kernel32.dll'. 
'VBSFix.exe' (Win32): Loaded 'C:\Windows\System32\KernelBase.dll'. 
'VBSFix.exe' (Win32): Loaded 'C:\Windows\System32\vcruntime140d.dll'. 
'VBSFix.exe' (Win32): Loaded 'C:\Windows\System32\ucrtbased.dll'. 
The thread 0x625c has exited with code 0 (0x0).
'VBSFix.exe' (Win32): Loaded 'C:\Windows\System32\sechost.dll'. 
'VBSFix.exe' (Win32): Loaded 'C:\Windows\System32\rpcrt4.dll'. 
'VBSFix.exe' (Win32): Loaded 'C:\Windows\System32\kernel.appcore.dll'. 
'VBSFix.exe' (Win32): Loaded 'C:\Windows\System32\msvcrt.dll'. 
The thread 0x649c has exited with code -7 (0xfffffff9).
The thread 0xa28 has exited with code -7 (0xfffffff9).
The program '[25436] VBSFix.exe' has exited with code -7 (0xfffffff9).

```
And a terminal of:
```
Standalone mode
PEB Base: 0x000000731971D000.
ImageBaseAddress: 0x00007FF6BCEC0000.
FindPattern failed.

E:\GitHub\RyzenMasterVBSFix\x64\Debug\VBSFix.exe (process 7188) exited with code -7.
To automatically close the console when debugging stops, enable Tools->Options->Debugging->Automatically close the console when debugging stops.
Press any key to close this window . . .
```

If it was pretty much any other language I could probably dig in an try things, but unfortunately, I don't know much about C++ or how applications made using it are supposed to work, let alone trying to debug them.

**Comment by milesburchell** [comment]
What version of Ryzen Master are you using? Could be that my pattern is not general enough. If it's the latest version, this issue is not replicable in my environment. I would suspect software interference (eg. Anti-virus)

**Comment by adamhl8** [comment]
I'm also getting `FindPattern failed.` Using verision `2.2.0.1543`.

**Comment by milesburchell** [comment]
Again, I can't replicate this issue on Windows 10 1909. I suspect that your security software is interfering with process interoperation. For those without the means to debug themselves, I pushed a new release with further debug output if FindPattern fails. It will tell you how many bytes were read from the target process. Should be 131,072 (0x20000 hex).

**Comment by Nightliss** [comment]
@milesburchell 

![image](https://user-images.githubusercontent.com/30467858/83733772-4dbb4580-a64e-11ea-9254-22f414faf362.png)

Win10 Pro Build 2004 and Ryzen Master 2.2.0.1543

**Comment by adamhl8** [comment]
I'm also on Win 10 Home Build 2004. Seems to be an issue with 2004 then?

![EgBbqM1](https://user-images.githubusercontent.com/1844269/83818903-76ffc400-a696-11ea-83f7-f8c81260c05c.png)

**Comment by MostHated** [comment]
Pro v2004 here as well

![](https://i.imgur.com/UTkW6Gn.png)

**Comment by milesburchell** [comment]
Seems likely v2004 has broken something. I am using the latest W10SDK for v2004, yet here we are. Unfortunately Microsoft's wonderful new update rollout system won't let me update regardless of "Insider" status.

**Comment by leolozes** [comment]
Same here, with Windows 2004 (19041.264)

PEB Base: 0x0000009C1B75F000.
ImageBaseAddress: 0x00007FF6684A0000.
FindPattern failed, 131072 bytes of memory were read.

**Comment by mmozeiko** [comment]
This has nothing to do with Windows 10 v2004. The executable simply is different in new Ryzen Master version. AMD Ryzen Master.exe v2.2.0.1543 does not have pattern 0x39 0x7D 0x90 0x0F 0x84 in .text segment. But it has bunch of places where 0x33 0xD2 0x48 0x8D 0x4C 0x24 0x60 can be found.

**Comment by mmozeiko** [comment]
This is how it looks like in v2.2.0.1543 executable:

![image](https://user-images.githubusercontent.com/1665010/84615435-dcfdff80-ae7d-11ea-9311-65cbb7f6144e.png)

If I replace that highlighted jz instruction with unconditional jump (bytes E9 F8 00 00 00) then it works for me.

**Comment by MostHated** [comment]
Well, for those of us who don't know C++ or how to replace instructions with an unconditional jump, what is the recommendation on getting this to work? Getting a different version of Ryzen Master?

**Comment by mmozeiko** [comment]
1) Download [HxD](https://mh-nexus.de/en/hxd/)
2) Copy "C:\Program Files\AMD\RyzenMaster\bin\AMD Ryzen Master.exe" to your Desktop
3) Launch HxD and open "AMD Ryzen Master.exe" from your Desktop
4) Choose `Search -> Goto` and enter `163e7`. Now you should see line like this (make values match):
![image](https://user-images.githubusercontent.com/1665010/84617751-c6a77200-ae84-11ea-81db-bd528e600aaa.png)
5) Now overwrite that part highlighted in red with E9 F8 00. Now it should look like this:
![image](https://user-images.githubusercontent.com/1665010/84617812-fb1b2e00-ae84-11ea-80c5-6e5e844d99b6.png)
6) Press Save and close HxD.
7) Now make sure you make copy of original exe file (just in case you needed it). Rename "AMD Ryzen Master.exe" to something like "AMD Ryzen Master.original.exe" in "C:\Program Files\AMD\RyzenMaster\bin" folder.
8) Move "AMD Ryzen Master.exe" from your desktop back to original folder. And then you can run it.

**Comment by MostHated** [comment]
Oh, doing it that way doesn't involve this project at all, yeah? Isn't that what would be considered the "normal" way of getting it working?

**Comment by mmozeiko** [comment]
Right. If you want to fix this in code, then you'll need modify C++ code.
I don't see point of doing this as next version of RyzenMaster will again change how how code looks like and again different offset/byte pattern will be needed. Then what's the point of change & compile code for every version from scratch, if just one time replacement for new version is all you need.

**Comment by MostHated** [comment]
Understandable, good point. I appreciate the info.

**Comment by MostHated** [comment]
> 
> 
> Right. If you want to fix this in code, then you'll need modify C++ code.
> I don't see point of doing this as next version of RyzenMaster will again change how how code looks like and again different offset/byte pattern will be needed. Then what's the point of change & compile code for every version from scratch, if just one time replacement for new version is all you need.

I finally got around to what you mentioned and it worked straight away. Thanks for that. 👍

**Comment by milesburchell** [comment]
> This is how it looks like in v2.2.0.1543 executable:
> 
> ![image](https://user-images.githubusercontent.com/1665010/84615435-dcfdff80-ae7d-11ea-9311-65cbb7f6144e.png)
> 
> If I replace that highlighted jz instruction with unconditional jump (bytes E9 F8 00 00 00) then it works for me.

The images you're showing bear no resemblance to the the .exe I have on disk for v2.2.0.1543. This is strange. Perhaps the installer is deploying different executables for different processor families or something? Can you upload your .exe for me to look at?

> Right. If you want to fix this in code, then you'll need modify C++ code.
> I don't see point of doing this as next version of RyzenMaster will again change how how code looks like and again different offset/byte pattern will be needed. Then what's the point of change & compile code for every version from scratch, if just one time replacement for new version is all you need.

The whole idea of the project is that the patterns should be universal, that's the idea. It works for me for both older and newer versions of Ryzen Master, but the plot thickens.

**Comment by milesburchell** [comment]
Or if anyone else who's been having issues could upload their AMD Ryzen Master.exe, that would be helpful.

**Comment by adamhl8** [comment]
[AMD Ryzen Master.zip](https://github.com/milesburchell/RyzenMasterVBSFix/files/4789628/AMD.Ryzen.Master.zip)

If it matters, I did what @mmozeiko suggested and that worked.

**Comment by mmozeiko** [comment]
> The images you're showing bear no resemblance to the the .exe I have on disk for v2.2.0.1543. This is strange. Perhaps the installer is deploying different executables for different processor families or something? Can you upload your .exe for me to look at?

I could believe that. Here's my original exe: https://gofile.io/d/o56Da3

I'm on Windows Pro v2004, Ryzen 3950X.

**Comment by mmozeiko** [comment]
I've unpacked my installer. Only one .exe was inside.

Here's how you do it:
```
mkdir C:\unpacked1
AMD-Ryzen-Master.exe /a /s /V"/qn TARGETDIR=C:\unpacked1"
mkdir C:\unpacked2
"C:\unpacked1\Packages\AMD Ryzen Master UI.exe" /a /s /V"/qn TARGETDIR=C:\unpacked2"
```
Now you will have `C:\unpacked2\Program Files 64\AMD\RyzenMaster\bin\AMD Ryzen Master.exe` file.
For me it matches that file I've uploaded in previous comment.

AMD-Ryzen-Master.exe I have was downloaded on June 4th. 137,814,432 bytes.
![image](https://user-images.githubusercontent.com/1665010/84840977-ec07bd80-aff5-11ea-9f3a-8d4c6e39e754.png)

**Comment by milesburchell** [comment]
There must be more than one inside because the .exe you uploaded is totally different to the .exe that the installer drops for me (Ryzen 2600). 800kb vs 1100kb, 2.2.0.1543 vs 1.5.4.1334. Presumably a newer version is distributed for Zen-2 based chips. Not entirely sure why AMD is doing it this way as opposed to just checking the system's chip, seems strange. I will update the pattern searching to support both versions when I can. Thanks

**Comment by milesburchell** [comment]
> I've unpacked my installer. Only one .exe was inside.
> 
> Here's how you do it:
> 
> ```
> mkdir C:\unpacked1
> AMD-Ryzen-Master.exe /a /s /V"/qn TARGETDIR=C:\unpacked1"
> mkdir C:\unpacked2
> "C:\unpacked1\Packages\AMD Ryzen Master UI.exe" /a /s /V"/qn TARGETDIR=C:\unpacked2"
> ```

This isn't unpacking the installer. It won't reveal any files the installer chooses not to install on your system.

**Comment by mmozeiko** [comment]
You're right. I did not realize that this will perform same logic as during installation.
After a bit more digging insides of msi/cab files I found three different `AMD Ryzen Master.exe` files. One matches what I have. And one is close to size you mentioned (1115KB). No idea what is third one. Here are all three of them: 
[amd_ryzen_master.zip](https://github.com/milesburchell/RyzenMasterVBSFix/files/4790485/amd_ryzen_master.zip)

There are lot of things in copies of three - AMDRyzenMasterDriver.sys driver, bunch of Qt dll/css/js files, etc...
